examples/medplum-chart-demo/src/bots/core/bot-utils.ts

```diff
@@ -0,0 +1,271 @@
+import { generateId, getReferenceString } from '@medplum/core';
+import {
+  BundleEntry,
+  BundleEntryRequest,
+  ClinicalImpression,
+  Coding,
+  Condition,
+  Encounter,
+  Observation,
+  ObservationComponent,
+  Patient,
+  Practitioner,
+  Quantity,
+  Reference,
+} from '@medplum/fhirtypes';
+
+interface ObservationData {
+  bloodPressure: {
+    systolic?: number;
+    diastolic?: number;
+  };
+  height?: Quantity;
+  weight?: Quantity;
+  bmi?: Quantity;
+  date?: string;
+}
+
+export interface GeneralObservationData extends ObservationData {
+  hotFlash?: boolean;
+  moodSwings?: boolean;
+  vaginalDryness?: boolean;
+  sleepDisturbance?: boolean;
+  selfReportedHistory?: string;
+}
+
+export interface GynecologyObservationData extends ObservationData {
+  lastPeriod?: string;
+  contraception?: Coding;
+  lastMammogram?: string;
+  smokingStatus?: Coding;
+  drugUse?: Coding;
+  housingStatus?: Coding;
+}
+
+export interface ObstetricObservationData extends ObservationData {
+  gravida?: number;
+  para?: number;
+  gestationalDays?: number;
+  gestationalWeeks?: number;
+  totalWeightGain?: Quantity;
+}
+
+export interface ConditionData {
+  reasonForVisit: Coding;
+  problemList: boolean;
+}
+
+export interface ClinicalImpressionData {
+  visitLength?: number;
+  assessment?: string;
+}
+
+export function createObservationEntries(
+  observationData: GeneralObservationData | GynecologyObservationData | ObstetricObservationData,
```

Comment:
Since they all extend `ObservationData`, you can just set the type to `ObservationData`